### PR TITLE
🧒🏻 Docker, pytest and cypress setups

### PR DESCRIPTION
**Pull request by HamzaAnis**
## Description :star2:
Docker setup has been changed. Workflows run on github hosted runner on every pull request to dev branch

Wrote E2E test case for
- Home Page
- Apply Page
- Login Page
- Forget Page
- Events Page
- Appointments Page
- Explore Page
- Sending and receiving messages and Notifications
- Your Videos Page
- Profile Page
- Admin Add new or download existing Accounts data

Backend test cases for 
- Authorization
- Check permissions of different users
- Add, Update and Delete Events
- Apply for different roles
- Find profiles of different roles
- Send and receive messages
- Check private users visibility
- Update profile info
- Add or delete Videos
- Add, remove images from imgur, google 
- Generate JWT token from firebase and Authorize 

All tests are formatted and should display relevant message on failure. After each run, the data generated by them such as events, appointments etc are cleaned up by them.

### CODE REVIEW

File: mentee test case/cypress/e2e/pages/HomePage.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+
+
+export class HomePage {
+    componentExist() {
+        cy.url().should('eq', 'https://mentee-dev.herokuapp.com/')
```
Comment (robert-dm) [inline mentee test case/cypress/e2e/pages/HomePage.js:5]: @HamzaAnis lets please avoid hard coding URLs as we will have another URL for PROD.

File: mentee test case/cypress/e2e/pages/HomePage.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        cy.get(':nth-child(2) > .ant-card').should('have.attr', 'class', 'ant-card ant-card-bordered css-ot64mz css-wxm1m1').and('contain', 'New').should('be.visible')
+        cy.get('span.anticon.anticon-right-circle').should('exist').and('have.attr', 'aria-label', 'right-circle').should('be.visible')
+        cy.get('.ant-card-meta-description').should('exist').and('contain', 'Apply - Train - Build').should('be.visible')
+
+        cy.get('.css-5lbmdi').should('exist').and('be.visible').and('have.attr', 'xmlns', 'http://www.w3.org/2000/svg')
+
+    }
+
+    isClickable() {
+        cy.url().should('eq', 'https://mentee-dev.herokuapp.com/')
```
Comment (robert-dm) [inline mentee test case/cypress/e2e/pages/HomePage.js:27]: @HamzaAnis remove hard coding of URL

File: mentee test case/cypress/e2e/test/ApplyPageTest.cy.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import { describe } from "mocha";
+
+describe('Apply Page', () => {
+    beforeEach('Open Apply Page', () => {
+        // TODO: Will add base url from config
+        cy.visit('https://mentee-dev.herokuapp.com/apply')
```
Comment (robert-dm) [inline mentee test case/cypress/e2e/test/ApplyPageTest.cy.js:6]: @HamzaAnis remove hard coding of URL

File: mentee test case/cypress/e2e/test/HomePageTest.cy.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import { HomePage } from "../pages/HomePage"
+
+const homePage = new HomePage()
+
+describe('HomePage', () => {
+    beforeEach('Open Home Page', () => {
+        cy.visit('https://mentee-dev.herokuapp.com/')
```
Comment (robert-dm) [inline mentee test case/cypress/e2e/test/HomePageTest.cy.js:7]: @HamzaAnis remove hard coding of URL

**Review by robert-dm — CHANGES_REQUESTED** [review]

File: mentee test case/cypress/e2e/test/HomePageTest.cy.js (hunk 4 above)
Comment (HamzaAnis) [reply mentee test case/cypress/e2e/test/HomePageTest.cy.js]: Resolved.

File: mentee test case/cypress/e2e/test/ApplyPageTest.cy.js (hunk 3 above)
Comment (HamzaAnis) [reply mentee test case/cypress/e2e/test/ApplyPageTest.cy.js]: Resolved.

File: mentee test case/cypress/e2e/pages/HomePage.js (hunk 2 above)
Comment (HamzaAnis) [reply mentee test case/cypress/e2e/pages/HomePage.js]: Resolved.

File: mentee test case/cypress/e2e/pages/HomePage.js (hunk 1 above)
Comment (HamzaAnis) [reply mentee test case/cypress/e2e/pages/HomePage.js]: @robert-dm Resolved.

**Review by robert-dm — APPROVED** [review]

**Review by robert-dm — APPROVED** [review]

File: backend/api/utils/.DS_Store
Comment (robert-dm) [inline backend/api/utils/.DS_Store:1]: @HamzaAnis please remove this file

File: backend/api/.DS_Store
Comment (robert-dm) [inline backend/api/.DS_Store:1]: @HamzaAnis please remove this file

**Review by robert-dm — CHANGES_REQUESTED** [review]
Please remove the two mentioned files @HamzaAnis

File: backend/api/utils/.DS_Store
Comment (HamzaAnis) [reply backend/api/utils/.DS_Store:1]: It is removed in this PR. It was present on the other branches.

**Pull request merged**